outputs/otlp_metrics.go

```diff
@@ -0,0 +1,237 @@
+// down the meter provider.
+func initMeterProvider(ctx context.Context, config *types.OTLPMetrics) (func(context.Context) error, error) {
+	var err error
+	var metricExporter sdkmetric.Exporter
+	switch protocol := config.Protocol; protocol {
+	case "grpc":
+		metricExporter, err = createGRPCExporter(ctx, config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create gRPC exporter: %v", err)
+		}
+	case "http/protobuf":
+		metricExporter, err = createHTTPExporter(ctx, config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create HTTP exporter: %v", err)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported OTLP transport protocol: %s", protocol)
+	}
+
+	res, err := sdkresource.New(ctx,
+		sdkresource.WithSchemaURL(semconv.SchemaURL),
+		sdkresource.WithAttributes(
+			semconv.ServiceName(serviceName),
+			semconv.ServiceVersion(serviceVersion),
+		),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create resource: %v", err)
+	}
+
+	meterProvider := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(sdkmetric.NewPeriodicReader(metricExporter)),
+		sdkmetric.WithResource(res),
+	)
+
+	otel.SetMeterProvider(meterProvider)
+	return meterProvider.Shutdown, nil
+}
+
+func createGRPCExporter(ctx context.Context, config *types.OTLPMetrics) (sdkmetric.Exporter, error) {
+	var options []otlpmetricgrpc.Option
+	if config.CheckCert == false {
+		options = append(options, otlpmetricgrpc.WithInsecure())
+	}
+
+	return otlpmetricgrpc.New(ctx, options...)
+}
+
+func createHTTPExporter(ctx context.Context, config *types.OTLPMetrics) (sdkmetric.Exporter, error) {
+	var options []otlpmetrichttp.Option
+	if config.CheckCert == false {
+		options = append(options, otlpmetrichttp.WithInsecure())
+	}
+	return otlpmetrichttp.New(ctx, options...)
+}
+
+const (
+	metricName                 = "falcosecurity_falco_rules_matches_total"
+	metricDescription          = "Number of times rules match"
+	metricAttributeUUIDKey     = "uuid"
+	metricAttributeSourceKey   = "source"
+	metricAttributePriorityKey = "priority"
+	metricAttributeRuleKey     = "rule"
+	metricAttributeHostnameKey = "hostname"
+	metricAttributeTagsKey     = "tags"
+)
+
+// OTLPMetricsPost generates a new OTEL metric data point for the provided falco payload.
+func (c *Client) OTLPMetricsPost(falcoPayload types.FalcoPayload) {
+	c.Stats.OTLPMetrics.Add(Total, 1)
+
+	meter := otel.Meter(meterName)
+	ruleCounter, err := meter.Int64Counter(metricName, metric.WithDescription(metricDescription))
+	if err != nil {
+		go c.CountMetric(Outputs, 1, []string{"output:otlpmetrics", "status:error"})
+		c.Stats.OTLPMetrics.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "otlpmetrics", "status": Error}).Inc()
+		log.Printf("[ERROR] : OTLP Metrics - Error generating metric: %v\n", err)
+		return
+	}
+
+	attrs := []attribute.KeyValue{
+		attribute.String(metricAttributeUUIDKey, falcoPayload.UUID),
```

Comment:
this is not useful, it will increase the cardinality for nothing, can you remove it please